xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp

```diff
@@ -185,17 +185,35 @@ COverlayTextureGL::COverlayTextureGL(CDVDOverlayImage* o)
 
   glBindTexture(GL_TEXTURE_2D, 0);
 
-  if (o->source_width && o->source_height)
+  if (o->source_width > 0 && o->source_height > 0)
   {
-    float center_x = (0.5f * o->width  + o->x) / o->source_width;
-    float center_y = (0.5f * o->height + o->y) / o->source_height;
-
-    m_align = ALIGN_VIDEO;
     m_pos = POSITION_RELATIVE;
-    m_x = center_x;
-    m_y = center_y;
-    m_width = static_cast<float>(o->width) / o->source_width;
-    m_height = static_cast<float>(o->height) / o->source_height;
+
+    m_x = (0.5f * o->width + o->x) / o->source_width;
+    m_y = (0.5f * o->height + o->y) / o->source_height;
+
+    if (o->source_height == static_cast<int>(rSource.Height()) ||
+      (static_cast<int>(rSource.Height()) % o->source_height == 0))
+    {
+      // We check also for multiple of source_height because for
+      // example 1080P subtitles can be used on 4K videos
+      m_align = ALIGN_VIDEO;
+      m_width = static_cast<float>(o->width) / o->source_width;
+      m_height = static_cast<float>(o->height) / o->source_height;
+    }
+    else
+    {
+      // If the source_height is higher than the video source height (rSource)
+      // means that the video has been remuxed/cropped (removed black bars).
```

Comment:
```suggestion
      // means that the video has been recoded/cropped (removed black bars).
```
Remux only does not remove black bars